tests/unit/updateTestResults.unitTest.ts

```diff
@@ -217,5 +216,40 @@ describe("updateTestResults", () => {
                 });
             });
         });
+
+        context("and when validating test types", () => {
+            context("and the test type contains a field that is not applicable", () => {
+                it("should return validation Error 400", () => {
+                    MockTestResultsDAO = jest.fn().mockImplementation();
+
+                    testResultsService = new TestResultsService(new MockTestResultsDAO());
+                    testToUpdate = testResultsMockDB[1];
+                    expect.assertions(3);
+                    return testResultsService.updateTestResult(testToUpdate.systemNumber, testToUpdate, msUserDetails)
+                      .catch((errorResponse: { statusCode: any; body: any; }) => {
+                          expect(errorResponse).toBeInstanceOf(HTTPError);
+                          expect(errorResponse.statusCode).toEqual(400);
+                          expect(errorResponse.body.errors[0]).toEqual(["\"prohibitionIssued\" is not allowed", "\"certificateNumber\" is not allowed"]);
```

Comment:
I was wondering if we could perhaps dynamically get "prohibitionIssued" and "certificateNumber" error message as they seem to be part of validation schema. In case if we had to change this field later in the future, the test wouldn't break. Thoughts?

Reply:
I think the only way to get them dynamically is to use the TestResultsService validateTestTypes function and compare the output of this function to the output of the updateTestResult function. Thoughts?

Reply:
resolving as we agreed to keep as it is. Since, it is external and wouldn't want to create possibly create false positive if we were to map the messages in a config locally perhaps.